packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/Choices.js

```diff
@@ -1,73 +1,123 @@
 import ChoiceButton from './components/ChoiceButton'
 
+const StyledGrid = styled(Box)`
+  display: grid;
+  grid-auto-flow: column;
+  grid-gap: 2px;
+  grid-template-rows: repeat(${props => props.rowsCount}, auto);
+`
+
 function Choices (props) {
   const {
+    autoFocus,
     filteredChoiceIds,
+    handleDelete,
     onChoose,
     selectedChoiceIds,
     task
   } = props
+  const [focusIndex, setFocusIndex] = useState(0)
+
+  useEffect(() => {
+    setFocusIndex(0)
+  }, [filteredChoiceIds])
 
   const columnsCount = howManyColumns(filteredChoiceIds)
-  const sortedFilteredChoiceIds = sortIntoColumns(filteredChoiceIds, columnsCount)
+  const rowsCount = Math.ceil(filteredChoiceIds.length / columnsCount)
   const thumbnailSize = task.alwaysShowThumbnails ? 'small' : whatSizeThumbnail(filteredChoiceIds)
 
+  function handleKeyDown (choiceId, event) {
+    const index = filteredChoiceIds.indexOf(choiceId)
+    let newIndex
+    switch (event.key) {
+      case 'ArrowDown': {
+        event.preventDefault()
+        event.stopPropagation()
+        newIndex = (index + 1) % filteredChoiceIds.length
+        setFocusIndex(newIndex)
+        return false
+      }
+      case 'ArrowUp': {
+        event.preventDefault()
+        event.stopPropagation()
+        newIndex = index - 1
+        if (newIndex === -1) {
+          newIndex = filteredChoiceIds.length - 1
+        }
+        setFocusIndex(newIndex)
+        return false
+      }
+      case 'Backspace': {
+        event.preventDefault()
+        event.stopPropagation()
+        handleDelete(choiceId)
+        return false
+      }
+      default: {
+        return true
+      }
+    }
+  }
+
   return (
-    // Could possibly drop box by styling gap color? Is that possible? Or using border on button?
-    <Box
+    <StyledGrid
       background={{
         dark: 'dark-1',
         light: 'light-1'
       }}
+      fill
+      rowsCount={rowsCount}
     >
-      <Grid
-        columns={{
-          count: columnsCount,
-          size: 'auto'
-        }}
-        fill
-        gap='2px'
-      >
-        {sortedFilteredChoiceIds.map((choiceId) => {
-          const choice = task.choices?.[choiceId] || {}
-          const selected = selectedChoiceIds.indexOf(choiceId) > -1
-          const src = task.images?.[choice.images?.[0]] || ''
-          return (
-            <ChoiceButton
-              key={choiceId}
-              choiceId={choiceId}
-              choiceLabel={choice.label}
-              onChoose={onChoose}
-              selected={selected}
-              src={src}
-              thumbnailSize={thumbnailSize}
-            />
-          )
-        })}
-      </Grid>
-    </Box>
+      {filteredChoiceIds.map((choiceId, index) => {
+        const choice = task.choices?.[choiceId] || {}
+        const selected = selectedChoiceIds.indexOf(choiceId) > -1
+        const src = task.images?.[choice.images?.[0]] || ''
+
+        const hasFocus = autoFocus && (index === focusIndex)
```

Comment:
One issue this is creating, is if filters are selected and the `filteredChoiceIds` changes, focus is grabbed from the filters DropButton content and sent to Choices (first ChoiceButton). Refactoring so Choices is aware of when filtering is happening will fix this issue, but there might be other refactoring of CharacteristicFilters, so noting this issue here, but plan to discuss more.

Reply:
If you have a plan for it, let's chat and it can be in another PR?